client/lib/graph/query.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { isString, uniqueId } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { clearRequests } from 'state/requests/actions';
+
+const query = ( mapPropsToQuery ) => ( WrappedComponent ) => {
+	return class GraphQueryComponent extends Component {
+		state = {
+			results: null
+		};
+
+		uid = uniqueId();
+
+		static contextTypes = {
+			graph: PropTypes.object
+		};
+
+		constructor( props, context ) {
+			super( props, context );
+			this.buildQuery( props );
+		}
+
+		componentDidMount() {
+			this.unsubscribe = this.context.graph.store.subscribe( () => {
+				this.request();
+			} );
+			this.request();
+		}
+
+		componentWillUnmount() {
+			this.unsubscribe && this.unsubscribe();
```

Comment:
never since componentDidMount will be called before, but It seems to me like a best practice to check all extra class attributes before using them. (I do the same for refs).

My logic here is : Here we know it's a React Component Class, so yeah certainly `componentDidMount` is called before but we think in a more OO way,  `componentDidMount` is not the constructor of our class, so I check first. 

But I don't mind droping it